lib/url.ts

```diff
@@ -1,8 +1,56 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-export interface URLQuery {
-  [queryParameterName: string]: string;
+export class URLQuery {
+  private readonly _rawQuery: { [queryParameterName: string]: string } = {};
+
+  public any(): boolean {
+    return Object.keys(this._rawQuery).length > 0;
+  }
+
+  public set(parameterName: string, parameterValue: string | undefined): void {
+    if (parameterName) {
+      if (parameterValue) {
+        this._rawQuery[parameterName] = parameterValue;
```

Comment:
any potential issues related to meaningful differences between an empty and an undefined parameter value?

Reply:
I thought about it and decided to keep things simple by treating an undefined parameter value the same as an empty parameter value (both mean a parameter that doesn't exist or won't exist anymore). If we change our minds in the future, it will be pretty easy to change this.

Reply:
I think AutoRest tests will wind up proving whichever way this needs to go.